applications/serial_lte_modem/src/slm_uart_handler.c

```diff
@@ -329,15 +331,26 @@ int slm_uart_power_off(void)
 	return err;
 }
 
-int slm_uart_configure(void)
+/* Configures UART according to the device tree. */
+static int slm_uart_configure(void)
 {
+	struct uart_config cfg;
 	int err;
 
-	LOG_DBG("Set UART baudrate to: %d", slm_uart.baudrate);
+	err = uart_config_get(uart_dev, &cfg);
+	if (err) {
+		LOG_ERR("uart_config_get: %d", err);
+		return err;
+	}
 
-	err = uart_configure(uart_dev, &slm_uart);
-	if (err != 0) {
+	err = uart_configure(uart_dev, &cfg);
```

Comment:
I don't think that you need to call this as the baudrate is already set. You just need to get it with uart_config_get to show the relevant information in traces. I think that this method could be removed and put back to slm_uart_handler_init, but can be kept as a separate method as well, as long as it is renamed to indicate that we are not changing anything.